### DISCUSSION TITLE
How to redirect to an external domain through `res.redirect()` from an API route

### DISCUSSION THREAD

**Discussion by miguderp** [vercel/next.js#28960] [Help] [answered]
Hi 👋🏻 ,

I am building an ecommerce and I use my API routes to generate a checkout URL (hosted on Mollie.com) and I am trying to directly redirect the user to the page through `res.redirect()` instead of returning said URL to my `fetch()` function and then redirect on the client-side.

Because it's on an external domain name, I get the classic CORS error message: 
`Blocking of a Cross-Origin Request: the “Same Origin” policy does not allow viewing of the remote resource located at https://www.mollie.com/paymentscreen/testmode/?method=creditcard&token=3.quxy0f. Reason: CORS request failed.`

`/pages/index.js`:
```
[...]
await fetch('/api/checkout', {
	headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
	credentials: 'include',
	method: 'POST',
	mode: 'cors',
	body: JSON.stringify(data),
}).then((response) => response.json())
[...]
```

`/pages/api/checkout`:
```
[...]
await mollieClient.payments.create({ ... }).then(async (payment) => {
	res.setHeader('Access-Control-Allow-Origin', 'https://www.mollie.com');
	res.setHeader('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
	res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, PATCH, DELETE');
	res.redirect(payment.getCheckoutUrl()); // ? returns a string such as https://www.mollie.com/paymentscreen/testmode/?method=creditcard&token=3.quxy0f
});
[...]
```

I have also tested through `next.config.js` headers directive but none seem to solve my issue – I think it’s probably due to my lack of knowledge in the matter but I'm out of ideas (and would really love to make this work this way).

Is it even possible?

Any ideas?
Thanks!

**Comment by vasco3** [comment]
did you figure it out? @miguderp 
I have the same problem

**Reply by miguderp** [reply]
Unfortunately not @vasco3, ended up going the traditional way for the time being and redirect on the client-side

**Reply by vasco3** [reply]
I'm doing the same

**Comment by miguderp** [comment, marked as answer]
Coming back to this just in case if someone else gets here through Google, the answer was `res.redirect(307, '/path-to-redirect');`
